src/Model_Hydro/MHD_Init_BField_ByFile.cpp

```diff
@@ -0,0 +1,369 @@
+#include "GAMER.h"
+
+#ifdef MHD
+#ifdef SUPPORT_HDF5
+#include "hdf5.h"
+
+static int nAx, nAy, nAz;
+static double Axmin, Aymin, Azmin;
+static double Adx, Ady, Adz;
+static double *Axcoord, *Aycoord, *Azcoord;
+
+
+double TSC_Weight( const double x );
+double VecPot_Interp( const double field[], const double xx, const double yy, 
+		      const double zz, const int fdims[], const int fbegin[] );
+void VecPot_ReadField( hid_t mag_file_id, const int ibegin, const int jbegin,
+                       const int kbegin, const int iend, const int jend,
+                       const int kend, double Ax[], double Ay[], double Az[] );
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  MHD_Init_BField_ByFile
+// Description :  Use the input uniform-mesh array stored in the file "B_Filename"
+//                to assign a magnetic vector potential to all real patches on level
+//                "B_lv" and take the curl of this potential to compute the magnetic field
+//
+// Note        :  
+//
+// Parameter   :  B_lv         : Target AMR level
+//
+// Return      :  amr->patch->fluid
+//-------------------------------------------------------------------------------------------------------
+void MHD_Init_BField_ByFile( const int B_lv )
+{
+
+#  ifndef MHD
+   Aux_Error( ERROR_INFO, "MHD must be enabled !!\n" );
+#  endif
+
+#  if SUPPORT_HDF5
+
+   if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Loading the magnetic field from the input file ...\n" );
+
+// set the number of OpenMP threads
+//#  ifdef OPENMP
+//   const int OMP_NThread = ( OPT__INIT_GRID_WITH_OMP ) ? OMP_NTHREAD : 1;
+//#  endif
+
+   const char B_Filename[] = "B_IC";
+
+   const double dh       = amr->dh[B_lv];
+
+   double *Axf, *Ayf, *Azf;
+
+   herr_t status;
+   hid_t dataset, dataspace;
+   hsize_t dims[3], maxdims[3];
+
+   if ( !Aux_CheckFileExist(B_Filename) )
+      Aux_Error( ERROR_INFO, "file \"%s\" does not exist !!\n", B_Filename );
+
+// Open the magnetic field file and determine the dimensionality of the vector
+// potential grid
+   hid_t mag_file_id = H5Fopen(B_Filename, H5F_ACC_RDONLY, H5P_DEFAULT);
+
+   if ( B_lv == 0 ) {
+
+     dataset = H5Dopen(mag_file_id, "magnetic_vector_potential_z", H5P_DEFAULT);
+     
+     dataspace = H5Dget_space(dataset);
+   
+     H5Sget_simple_extent_dims(dataspace, dims, maxdims);
+
+     H5Sclose(dataspace);
+     H5Dclose(dataset);
+
+     nAx = dims[0];
```

Comment:
Just to be sure, here we have assumed that the arrays like `magnetic_vector_potential_x` are stored in the format `[nAx][nAy][nAz]` instead of `[nAz][nAy][nAx]` in the row-major order, right? If yes, we should add comments on that for clarification since all other arrays in GAMER are stored in the format `[nz][ny][nx]`.